cartographer/cloud/internal/handlers/add_trajectory_handler.cc

```diff
@@ -60,10 +60,15 @@ void AddTrajectoryHandler::OnRequest(
     // Ignore initial poses in trajectory_builder_options.
     trajectory_builder_options.clear_initial_trajectory_pose();
 
-    GetContext<MapBuilderContextInterface>()
-        ->local_trajectory_uploader()
-        ->AddTrajectory(request.client_id(), trajectory_id, expected_sensor_ids,
-                        trajectory_builder_options);
+    if (!GetContext<MapBuilderContextInterface>()
+             ->local_trajectory_uploader()
+             ->AddTrajectory(request.client_id(), trajectory_id,
+                             expected_sensor_ids, trajectory_builder_options)) {
+      LOG(ERROR) << "Failed to create trajectory in uplink: " << trajectory_id;
+      Finish(::grpc::Status(::grpc::INTERNAL,
```

Comment:
Is this covered in the integration test?

Reply:
this is slightly tricky. I will do this in another PR.